modules/programs/zplug.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.zplug;
+
+  pkg = pkgs.srcOnly {
+    name = "zplug";
+    src = builtins.fetchGit {
```

Comment:
This is not reproducible and it also introduces a package to Home Manager. I much rather you'd add `zplug` as a package in https://github.com/NixOS/nixpkgs/ as a proper package with version, license information, and such. Once it is added to Nixpkgs you can refer to it through `pkgs.zplug`.